### PR TITLE
PROD 541: Chompmas streaks mystery boxes

### PR DESCRIPTION
**Pull request by samott**
Implements whitelisted Chompmas streaks mystery boxes.

Requires the mechanism update to function.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -37,8 +37,7 @@ OVERRIDE_USER_ID=
...
 
-# mystery box feature flag
-NEXT_PUBLIC_FF_MYSTERY_BOX_CLAIM_ALL=
-
 # bot feature flag
-FF_CHOMP_BOT=
\ No newline at end of file
+FF_CHOMP_BOT=
+
+CHOMPMAS_MIN_STREAK=7
```
Comment (marvinmarnold) [inline .env.example:43]: Add to Vercel and .env.example

**Comment by marvinmarnold** [comment]
I set CHOMPMAS_MIN_STREAK=1 in preview, but I'm not getting a mystery box
Screenshot 2024-12-11 at 4 23 58 PM

**Comment by marvinmarnold** [comment]
Oh I need to add myself to allowlist. I'll try that.

File: README.md [hunk 2]
```diff
@@ -175,3 +175,18 @@ yarn dev:restore-db
...
+
+## Add Users in MysteryBox Allowlist
+
+1. Add a csv file in the scripts folder (only addresses, no header)
+2. Make sure the name of file is `allowlist.csv`
+3. Verify the DATABASE_URL env
+
+**Run the following command**
+```sh
+yarn dev:add-mystery-box
```
Comment (marvinmarnold) [inline README.md:187]: package.json uses: dev:add-allowlist-users

**Comment by samott** [comment]
We also need the mechanism-engine update.

**Comment by marvinmarnold** [comment]
> We also need the mechanism-engine update.

We also need to run migrations for this to work. For now I just want to be able to test in staging so I want to merge. Can you:
1. Use the existing event type for claim all instead of chompas?
2. Double check that this code will not impact prod if merged (ie hidden behind FF feature flag)

**Comment by samott** [comment]
> > We also need the mechanism-engine update.
> 
> We also need to run migrations for this to work. For now I just want to be able to test in staging so I want to merge. Can you:
> 
>     1. Use the existing event type for claim all instead of chompas?

Done

>     2. Double check that this code will not impact prod if merged (ie hidden behind FF feature flag)

I added an extra flag. Everything looks good with it switched off.

**Pull request merged**